var/da/da_main/da_update_firstguess.inc

```diff
@@ -585,17 +588,37 @@ subroutine da_update_firstguess(grid, out_filename)
                              start_index,end_index1,               & !pat
                              ierr                                 )
 
-! Update moist potential temperature: for now, it is identical to T with use_theta_m=0.
-     rmse_var='THM'
-     call ext_ncd_get_var_info (dh1,trim(rmse_var),ndim1,ordering,staggering, &
-          start_index,end_index1, WrfType, ierr    )
-     call ext_ncd_write_field(dh1,DateStr1,TRIM(rmse_var),         &
-                             field3,WRF_REAL,0,0,0,ordering,       &
-                             staggering, dimnames ,                &
-                             start_index,end_index1,               & !dom
-                             start_index,end_index1,               & !mem
-                             start_index,end_index1,               & !pat
-                             ierr                                 )
+!
+! Update moist potential temperature perturbation THM
+!
+      rmse_var='THM'
+      call ext_ncd_get_var_info (dh1,trim(rmse_var),ndim1,ordering,staggering, &
+           start_index,end_index1, WrfType, ierr    )
+
+      if (grid%use_theta_m == 1) then ! convert dry theta perturbation to moist one
+         write(unit=message(1),fmt='(A, I2)') 'convert T to THM when use_theta_m = ', grid%use_theta_m
+         call da_message(message(1:1))
+         do k= 1,nsig_regional
+            do j= 1,nlat_regional
+               do i= 1,nlon_regional
+                 qvf = 1.+(R_v/R_d)*grid%moist(i,j,k,p_qv)
```

Comment:
@liujake 
Jake,
Is grid%moist the size of the computational region that is decomposed on a particular MPI rank?  Are the n{lat,lon}_regional dimensions just the computational size without halo zones?

Reply:
@liujake 
Jake,
The write seems to be using "if(rootproc)" as if the whole domain is output, so I am confused about the moist dimensions.

Also, I looked in this routine, and you don't do computations anywhere else. Perhaps the moist theta computation should not be done in this output routine? I am not a DA person, so I am not familiar with the conventions.

Reply:
@davegill Thanks for catching this! You are right. n{lat,lon}_regional dimensions are domain size. I adjusted qvf calculation right after patch_to_global for QVAPOR and update THM before updating QVAPOR. This way, I do not have to define another 3D field array or collect QVAPOR twice. Test passed.